ImageWidgets.go

```diff
@@ -107,51 +116,36 @@ func (i *ImageWidget) Build() {
 		mousePos := GetMousePos()
 
 		if cursorPos.X <= mousePos.X && cursorPos.Y <= mousePos.Y &&
-			cursorPos.X+int(i.width) >= mousePos.X && cursorPos.Y+int(i.height) >= mousePos.Y {
+			cursorPos.X+int(size.X) >= mousePos.X && cursorPos.Y+int(size.Y) >= mousePos.Y {
 			i.onClick()
 		}
 	}
 
 	imgui.ImageV(i.texture.tex.ID, size, i.uv0, i.uv1, ToVec4Color(i.tintColor), ToVec4Color(i.borderColor))
 }
 
-type imageState struct {
-	loading bool
-	failure bool
-	cancel  ctx.CancelFunc
-	texture *Texture
-	img     *image.RGBA
-}
-
-// Dispose cleans imageState (implements Disposable interface).
-func (is *imageState) Dispose() {
-	is.texture = nil
-	// Cancel ongoing image downloading
-	if is.loading && is.cancel != nil {
-		is.cancel()
-	}
-}
-
 var _ Widget = &ImageWithRgbaWidget{}
 
 // ImageWithRgbaWidget wraps ImageWidget.
 // It is more useful because it doesn't make you to care about
 // imgui textures. You can just pass golang-native image.Image and
 // display it in giu.
 type ImageWithRgbaWidget struct {
-	id   ID
-	rgba *image.RGBA
-	img  *ImageWidget
+	id  ID
+	rbt *ReflectiveBoundTexture
+	img *ImageWidget
 }
 
 // ImageWithRgba creates ImageWithRgbaWidget.
 // The default size is the size of the image,
 // to set a specific size, use .Size(width, height).
-func ImageWithRgba(rgba image.Image) *ImageWithRgbaWidget {
```

Comment:
I think that to keep giu as imple as possible, I'd suggest to auto-create this second variable and sotre it in imageState (as it was with *Texture)
If you think that user might want to have access to it, add a getter to ImageWithXXXWidget.

Reply:
see https://github.com/AllenDang/giu/pull/855#issuecomment-2379607305